### ISSUE TITLE
hive_generator ^1.0.0 requires build ^1.5.2

### ISSUE THREAD

**Issue by zs-dima** [hivedb/hive#605]
In case of updating json_serializable to 4.0.3 or etc exception appear:

```
Because no versions of hive_generator match >1.0.0 <2.0.0 and hive_generator 1.0.0 depends on build ^1.5.2, hive_generator ^1.0.0 requires build ^1.5.2.
And because json_serializable >=4.0.3 depends on build ^2.0.0, hive_generator ^1.0.0 is incompatible with json_serializable >=4.0.3.
So, because workshop depends on both json_serializable ^4.0.3 and hive_generator ^1.0.0, version solving failed.
pub get failed (1; So, because workshop depends on both json_serializable ^4.0.3 and hive_generator ^1.0.0, version solving failed.)
```

**Comment by MahmoudAbunaji** [comment]
same issue

**Comment by vandir** [comment]
#597 Just look at the latest open issues instead of opening a new one

**Comment by zs-dima** [comment]
@vandir #597 had been closed but issue had not been fixed!

**Comment by vandir** [comment]
If anyone has the same problem please replace the following lines:

```
dev_dependencies:
  hive_generator: ^1.0.0
  build_runner: ^1.12.1
```

with:
```
dev_dependencies:
  hive_generator:
  build_runner:
```

**Comment by zs-dima** [comment]
@vandir 
dev_dependencies is very bad and dangerous workaround - and is not really issue fix!

**Comment by vandir** [comment]
This is not a `dependency_overrides`, I've just removed the versions from `dev_dependencies`. Anyway it's just a temp fix. The alternative is to import the last commit directly from github or wait for the next update.

**Comment by DarkHeros09** [comment]
hello, any progress on this issue?

**Comment by TheMisir** [comment]
Currently I would suggest using `dependency_overrides` to temporarily suppress this issue. There was dependency incompatibility issues between some packages that was made by dart team which hive_generator uses for code generation. So that's why some dependencies are incompatible.

**Comment by TheMisir** [comment]
> @vandir
> dependency_overrides is very bad and dangerous workaround - and is not really issue fix!
> and packages downgrade is not solution as well!

If something works, it's a solution. That's how all the softwares in the world released. Otherwise it's not possible to complete any project without doing workarounds like that. [Perfect is the enemy of good.](https://en.wikipedia.org/wiki/Perfect_is_the_enemy_of_good)

**Comment by qbait** [comment]
Any ETA on that by any chance?
I needed to downgrades a lot of dependencies on my project because of the hive generator and with null safety turned on it's pretty painful :)

**Comment by kirill-plotnikov** [comment]
> > @vandir
> > dependency_overrides is very bad and dangerous workaround - and is not really issue fix!
> > and packages downgrade is not solution as well!
> 
> If something works, it's a solution. That's how all the softwares in the world released. Otherwise it's not possible to complete any project without doing workarounds like that. [Perfect is the enemy of good.](https://en.wikipedia.org/wiki/Perfect_is_the_enemy_of_good)

Unfortunately this doesn't work, if at least one another package require `build` ^2.0.0. ☹️

**Comment by return-main** [comment]
> Any ETA on that by any chance?
> I needed to downgrades a lot of dependencies on my project because of the hive generator and with null safety turned on it's pretty painful.
> 
> Suggestion how to temporarily solve it and keep the rest of my dependencies updated, very welcome :)

The fix is already available on [GitHub](https://github.com/hivedb/hive/commit/0a66d8668d199a09a82c2d49a3d914fef7cb14c5), so change your pubspec.yaml dev_dependencies to:
```yaml
dev_dependencies:
  hive_generator:  # ^1.0.0 with a dependency fix
    git:
      url: https://github.com/hivedb/hive.git
      path: hive_generator/
      ref: 0a66d8668d199a09a82c2d49a3d914fef7cb14c5
```

**Comment by TheMisir** [comment]
Fixed on 1.0.1

```yaml
dev_dependencies:
  hive_generator: ^1.0.1
```
